grid-client/state/state.go

```diff
@@ -322,6 +330,24 @@ func (st *State) LoadNetworkFromGrid(name string) (znet workloads.ZNet, err erro
 	znet.NodesIPRange = nodesIPRange
 	znet.Keys = keys
 	znet.WGPort = wgPort
+	if metdata.UserAcessIP != "" {
+		ip, err := gridtypes.ParseIPNet(metdata.UserAcessIP)
+		if err != nil {
+			return znet, errors.Wrapf(err, "failed to parse user access ip from metadata string %s", metdata.UserAcessIP)
+		}
+		znet.ExternalIP = &ip
+	}
+
+	if metdata.PrivateKey != "" {
+		key, err := wgtypes.ParseKey(metdata.PrivateKey)
+		if err != nil {
+			return znet, errors.Wrapf(err, "failed to parse user access private key from metadata string %s", metdata.PrivateKey)
+		}
+		znet.ExternalSK = key
+	}
+
+	znet.PublicNodeID = metdata.PublicNodeID
+	znet.AddWGAccess = (metdata.PrivateKey != "")
```

Comment:
I think you mentioned that we can't guarantee that AddWGAccess is true from the private key.
also we have the private key in the zos workload

Reply:
this is correct for private keys of network workloads, you can't conclude that a user requested wg access from the presence of a private key in a network workloads, because it's always there. but the private key stored in metadata is a different one, it's the key assigned to a user whenever they request wg access. so basically, the presence of a private key or an ip in the metadata indicates that the user requested wg access before.